irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -250,27 +274,36 @@ namespace iroha {
         std::shared_ptr<shared_model::interface::CommonObjectsFactory> factory,
         KeyValueStorage &block_store,
         std::shared_ptr<PendingTransactionStorage> pending_txs_storage,
-        std::shared_ptr<shared_model::interface::BlockJsonConverter> converter)
+        std::shared_ptr<shared_model::interface::BlockJsonConverter> converter,
+        std::shared_ptr<shared_model::interface::QueryResponseFactory>
+            response_factory)
         : sql_(sql),
           block_store_(block_store),
-          factory_(std::move(factory)),
+          common_objects_factory_(std::move(factory)),
           pending_txs_storage_(std::move(pending_txs_storage)),
           converter_(std::move(converter)),
+          query_response_factory_{std::move(response_factory)},
           log_(logger::log("PostgresQueryExecutorVisitor")) {}
 
     void PostgresQueryExecutorVisitor::setCreatorId(
         const shared_model::interface::types::AccountIdType &creator_id) {
       creator_id_ = creator_id;
     }
 
-    QueryResponseBuilderDone PostgresQueryExecutorVisitor::operator()(
+    void PostgresQueryExecutorVisitor::setQueryHash(
+        const shared_model::crypto::Hash &query_hash) {
```

Comment:
You can use shared_model::interface::types::HashType, so that you do not depend on cryptography directly